js/kucoinfutures.js

```diff
@@ -0,0 +1,834 @@
+                    '200004': InsufficientFunds,
+                    '230003': InsufficientFunds, // {"code":"230003","msg":"Balance insufficient!"}
+                    '260100': InsufficientFunds, // {"code":"260100","msg":"account.noBalance"}
+                    '400001': AuthenticationError, // Any of KC-API-KEY, KC-API-SIGN, KC-API-TIMESTAMP, KC-API-PASSPHRASE is missing in your request header.
+                    '400002': InvalidNonce, // KC-API-TIMESTAMP Invalid -- Time differs from server time by more than 5 seconds
+                    '400003': AuthenticationError, // KC-API-KEY not exists
+                    '400004': AuthenticationError, // KC-API-PASSPHRASE error
+                    '400005': AuthenticationError, // Signature error -- Please check your signature
+                    '400006': AuthenticationError, // The IP address is not in the API whitelist
+                    '400007': AuthenticationError, // Access Denied -- Your API key does not have sufficient permissions to access the URI
+                    '404000': NotSupported, // URL Not Found -- The requested resource could not be found
+                    '400100': BadRequest, // Parameter Error -- You tried to access the resource with invalid parameters
+                    '411100': AccountSuspended, // User is frozen -- Please contact us via support center
+                    '500000': ExchangeNotAvailable, // Internal Server Error -- We had a problem with our server. Try again later.
+                },
+                // 'broad': {
+                //     'Exceeded the access frequency': RateLimitExceeded,
+                //     'require more permission': PermissionDenied,
+                // },
+            },
+            'fees': {
+                'trading': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'taker': 0.001,
+                    'maker': 0.001,
+                },
+                'funding': {
+                    'tierBased': false,
+                    'percentage': false,
+                    'withdraw': {},
+                    'deposit': {},
+                },
+            },
+            'commonCurrencies': {
+                'HOT': 'HOTNOW',
+                'EDGE': 'DADI', // https://github.com/ccxt/ccxt/issues/5756
+                'WAX': 'WAXP',
+                'TRY': 'Trias',
+                'VAI': 'VAIOT',
+            },
+            'timeframes': {
+                '1m': 1,
+                '3m': undefined,
+                '5m': 5,
+                '15m': 15,
+                '30m': 30,
+                '1h': 60,
+                '2h': 120,
+                '4h': 240,
+                '6h': undefined,
+                '8h': 480,
+                '12h': 720,
+                '1d': 1440,
+                '1w': 10080,
+            },
+            'options': {
+                // 'version': 'v2',
+                // 'symbolSeparator': '-',
```

Comment:
Unless I comment these lines out, I receive `kucoinfutures Not Found`

Reply:
That is an inheritance conflict, i think.

> I receive kucoinfutures Not Found

From where exactly do you receive that? We should debug it.

Reply:
When I run `kucoinfutures fetchMarkets`, if I change version to `v1` I don't get this problem

```
% kucoinfutures fetchMarkets
2021-12-08T03:36:37.396Z
Node.js: v14.17.0
CCXT v1.63.10
[NotSupported] kucoinfutures Not Found

    at throwExactlyMatchedException  ../../ccxt/ccxt/js/base/Exchange.js:593        throw new exact[string] (message)                                               
    at handleErrors                  ../../ccxt/ccxt/js/kucoin.js:2364              this.throwExactlyMatchedException (this.exceptions['exact'], errorCode, this.id…
    at                               ../../ccxt/ccxt/js/base/Exchange.js:670        this.handleErrors (response.status, response.statusText, url, method, responseH…
    at processTicksAndRejections     internal/process/task_queues.js:95                                                                                             
    at async timeout                 ../../ccxt/ccxt/js/base/functions/time.js:203  return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetchMarkets                  ../../ccxt/ccxt/js/kucoinfutures.js:278        const response = await this.futuresPublicGetContractsActive (params);           
    at loadMarketsHelper             ../../ccxt/ccxt/js/base/Exchange.js:740        const markets = await this.fetchMarkets (params)                                
    at async main                    ../../ccxt/ccxt/examples/js/cli.js:208         await exchange.loadMarkets ()                                                   

kucoinfutures Not Found
```

Reply:
@samgermain please paste your verbose output (add --verbose to the cli call), that will show which endpoint is not found.

Reply:
Looks like it's this method failing `futuresPublicGetContractsActive`, but you have to see the verbose output and the URL it uses to understand exactly why.